src/Features/Core/Portable/NavigateTo/NavigateToItem.cs

```diff
@@ -0,0 +1,272 @@
+            DeclaredSymbolInfoIsNonNestedNamedType = declaredSymbolInfoIsNonNestedNamedType;
+            Kind = kind;
+            MatchKind = matchKind;
+            IsCaseSensitive = isCaseSensitive;
+            NameMatchSpans = nameMatchSpans;
+            SecondarySort = secondarySort;
+        }
+
+        public bool TryCreateSearchResult(Solution solution, [NotNullWhen(true)] out INavigateToSearchResult? result)
+        {
+            var document = solution.GetDocument(DocumentId);
+            if (document == null)
+            {
+                var project = solution.GetProject(DocumentId.ProjectId);
+                if (project != null)
+                    document = project.TryGetSourceGeneratedDocumentForAlreadyGeneratedId(DocumentId);
+            }
+            Contract.ThrowIfTrue(document == null && !IsStale, "We should always be able to map back a non-stale result to the solution");
+
+            if (document == null)
+            {
+                result = null;
+                return false;
+            }
+
+            result = new NavigateToSearchResult(this, document);
+            return true;
+        }
+
+        private class NavigateToSearchResult : INavigateToSearchResult, INavigableItem
+        {
+            private static readonly char[] s_dotArray = { '.' };
+
+            private readonly RoslynNavigateToItem _item;
+            private readonly Document _document;
+
+            public NavigateToSearchResult(RoslynNavigateToItem item, Document document)
+            {
+                _item = item;
+                _document = document;
+            }
+
+            string INavigateToSearchResult.AdditionalInformation
+            {
+                get
+                {
+                    //private static string ComputeAdditionalInfo(string documentName, string combinedProjectName, DeclaredSymbolInfo info)
+                    {
+                        // For partial types, state what file they're in so the user can disambiguate the results.
+                        if (_item.DeclaredSymbolInfoIsPartial)
+                        {
+                            return _item.DeclaredSymbolInfoIsNonNestedNamedType
+                                ? string.Format(FeaturesResources._0_dash_1, _document.Name, _item.CombinedProjectName)
+                                : string.Format(FeaturesResources.in_0_1_2, _item.DeclaredSymbolInfoContainerDisplayName, _document.Name, _item.CombinedProjectName);
+                        }
+                        else
+                        {
+                            return _item.DeclaredSymbolInfoIsNonNestedNamedType
+                                ? string.Format(FeaturesResources.project_0, _item.CombinedProjectName)
+                                : string.Format(FeaturesResources.in_0_project_1, _item.DeclaredSymbolInfoContainerDisplayName, _item.CombinedProjectName);
+                        }
+                    }
+                }
+            }
```

Comment:
this logic moved from teh main searcher.  It depends on actual Document/Project info, so we don't want to do it in OOP, where i want NavigateTo to not depend on those.  Instead, it does it on the host side before passing back out to vs.